base/range_iterator.hpp

```diff
@@ -0,0 +1,101 @@
+#include "std/iterator.hpp"
+#include "std/type_traits.hpp"
+
+namespace base
```

Comment:
```
y@y-thinkpad:~/hdd/coding/omim/base$ gg 'namespace my' | wc -l
34
y@y-thinkpad:~/hdd/coding/omim/base$ gg 'namespace base' | wc -l
3
```
Please, use namespace "my". Don't create new namespaces just because you like to introduce new names. I agree that all stuff in base/ must be in "namespace base", and it's worth to do that, but in a separate PR. For now, follow conventions in the directory.